emails/models.py

```diff
@@ -665,7 +667,11 @@ def save(self, *args, **kwargs) -> None:
                 locked_profile = Profile.objects.select_for_update().get(user=self.user)
                 check_user_can_make_another_address(locked_profile)
                 while True:
-                    if valid_address(self.address, self.domain_value):
+                    address_is_blocklisted = is_blocklisted(self.address)
+                    if (
+                        valid_address(self.address, self.domain_value)
+                        and not address_is_blocklisted
+                    ):
```

Comment:
Nit: either use variable assignment twice, or add to if clause twice. Here's the variable assign twice:

```suggestion
                    address_is_valid = valid_address(self.address, self.domain_value)
                    address_is_allowed = not is_blocklisted(self.address)
                    if address_is_valid and address_is_allowed:
```

Reply:
Done in https://github.com/mozilla/fx-private-relay/pull/4254/commits/f2e06e88b00cee3dcda4e07313516f1bd91902f4